### PR TITLE
Add new action pmpro_before_commit_express_checkout

### PR DESCRIPTION
**Pull request by mircobabini**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This one allows performing actions on order status = token, just before sending the user to the gateway to complete the payment.

Expecially useful to save pmpro_signup_username / pmpro_signup_password / pmpro_signup_email into order metas, to be used when trying to recover checkout failure if the browser session is lost, as reported #1338.

To be combined with this snippet: https://gist.github.com/mircobabini/bf9b65010124e4ab1e04c0c634681364

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: classes/gateways/class.pmprogateway_paypalexpress.php [hunk 1]
```diff
@@ -598,6 +598,15 @@ function setExpressCheckout(&$order)
 				//update order
 				$order->saveOrder();
 
+				/**
+				 * Allow performing actions just before sending the user to the gateway to complete the payment.
+				 *
+				 * @since TBD
```
Comment (ideadude) [inline classes/gateways/class.pmprogateway_paypalexpress.php:604]:
```suggestion
				 * @since 2.6.5
```

**Comment by ideadude** [comment]
The password saved to session in plaintext. So there are some security considerations in your gist around saving that into the DB in plaintext as well. I wonder if you could hash it and save that way. Could maybe use the wp_pre_insert_user_data to swap your original hashed version in for the doubly hashed version that the call to wp_insert_user would generate.

Despite that, the filter itself is useful. Merging in and pushing out in the 2.6.5 release.

**Comment by mircobabini** [comment]
> Could maybe use the wp_pre_insert_user_data to swap your original hashed version in for the doubly hashed version that the call to wp_insert_user would generate.

It's definitely possible a nice idea; gonna update it soon.

**Comment by mircobabini** [comment]
> Could maybe use the wp_pre_insert_user_data to swap your original hashed version in for the doubly hashed version that the call to wp_insert_user would generate.

Script updated: https://gist.github.com/mircobabini/bf9b65010124e4ab1e04c0c634681364

**Pull request merged**